Automattic/wp-calypso
# Plans: Remove unused price components

#### Changes proposed in this Pull Request

This PR removes a couple of unused price components:

* `plan-price-free` - unused since #49474
* `wpcom-plan-price` - unused since #24483

#### Testing instructions

Verify the removed components are not in use. There should be no bundle size changes.
